server/chat-plugins/sample-teams.ts

```diff
@@ -0,0 +1,515 @@
+			} else if (!file.teams[toID(query[0])] || !Object.keys(file.teams[toID(query[0])]).length ||
+				(Object.keys(file.teams[toID(query[0])].uncategorized).length &&
+					!Object.keys(file.teams[toID(query[0])]).filter(x => x !== 'uncategorized').length)) {
+				const name = Dex.formats.get(query[0]).exists ? Dex.formats.get(query[0]).name : toID(query[0]);
+				return `${buf}<p>No teams for ${name} were found.</p></div>`;
+			} else if (!query[1] || (!SampleTeams.findCategory(query[0], query[1]) && query[1] !== 'allteams')) {
+				buf += `<h3>Pick a category</h3><ul>`;
+				for (const category of Object.keys(file.teams[toID(query[0])])) {
+					buf += `<li><a class="button" style="text-decoration:inherit;color:inherit" target="replace" href="view-sampleteams-view-${query[0]}-${toID(category)}">${category}</button></li>`;
+				}
+				buf += `<li><a class="button" style="text-decoration:inherit;color:inherit" target="replace" href="view-sampleteams-view-${query[0]}-allteams">ALL</button></li></ul>`;
+			} else if (query[1] === 'allteams') {
+				buf += `<h3>All teams for ${SampleTeams.getFormatName(query[0])}</h3>`;
+				for (const categoryName in file.teams[toID(query[0])]) {
+					const category = file.teams[toID(query[0])][categoryName];
+					if (!Object.keys(category).length) continue;
+					buf += `<details><summary><h4 style="display:inline">${categoryName}</h4></summary>`;
+					for (const teamName in category) {
+						const team = category[teamName];
+						if (SampleTeams.checkPermissions(user, file.whitelist[query[0]] || [])) {
+							buf += `<button class="button" style="float:right" name="send" value="/sampleteams remove ${query[0]},${categoryName},${teamName}">Delete team</button>`;
+						}
+						buf += SampleTeams.formatTeam(teamName, team);
+					}
+					buf += `</details>`;
+					const index = Object.keys(file.teams[toID(query[0])]).indexOf(categoryName);
+					if (index !== Object.keys(file.teams[toID(query[0])]).length - 1) buf += `<hr />`;
+				}
+			} else if (SampleTeams.findCategory(query[0], query[1])) {
+				const categoryName = SampleTeams.findCategory(query[0], query[1])!;
+				buf += `<h3>Sample teams for ${SampleTeams.getFormatName(query[0])} in the ${categoryName} category</h3>`;
+				for (const teamName in file.teams[query[0]][categoryName]) {
+					const team = file.teams[query[0]][categoryName][teamName];
+					buf += SampleTeams.formatTeam(teamName, team);
+				}
+			}
+			buf += `</div>`;
+			return buf;
+		},
+		add(query, user, connection) {
+			this.title = `Sample Teams`;
+			const formats = Array.from(new Set(Dex.formats.all().map(format => (format.name.includes('(') ?
+				format.name.slice(0, format.name.indexOf('(')) : format.name).trim()))).map(formatName =>
+				Dex.formats.get(formatName)).filter(format =>
+				!format.name.includes('Custom') && format.effectType === 'Format' && !format.team &&
+				SampleTeams.checkPermissions(user, SampleTeams.whitelistedRooms(format.id) || []));
```

Comment:
This is terrifying. And difficult to read. Please use multiple variables to do this in a few lines?

Reply:
At the very least indent this in a more readable way...

Reply:
For some reason the linter would not let me indent more smoothly than this and would even give errors when parentheses were put on separate lines from the function

```js
likeThis.forAn(
  'example'
};